src/main/java/com/shapesecurity/shift/es2017/semantics/visitor/Visitor.java

```diff
@@ -0,0 +1,25 @@
+package com.shapesecurity.shift.es2017.semantics.visitor;
+
+import com.shapesecurity.functional.data.ImmutableList;
+import com.shapesecurity.functional.data.Monoid;
+import com.shapesecurity.shift.es2017.semantics.asg.Node;
+
+import javax.annotation.Nonnull;
+
+public final class Visitor extends MonoidalReducer<ImmutableList<Node>> {
+
+	public static ImmutableList<Node> getChildren(@Nonnull Node node) {
```

Comment:
Can we call this `getDescendants` or something? I usually assume that "children" is just direct children.